src/plugins/__init__.py

```
@@ -0,0 +1,48 @@
+        if name not in self._plugins:
+            raise ValueError(f"Plugin '{name}' not found. Available: {list(self._plugins.keys())}")
+        return self._plugins[name]
+
+    def list_plugins(self) -> Dict[str, Type[DataSource]]:
+        return self._plugins.copy()
+
+    def auto_discover(self, module_path: str = "src.plugins") -> None:
+        plugin_modules = [
+            'openaq',
+            'purpleair',
+            'waqi'
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)